client/src/main/java/com/paypal/selion/reports/reporter/excelreport/TestCaseResult.java

```diff
@@ -91,18 +95,70 @@ public void setITestResultobj(ITestResult iTestResult) {
             this.setGroup(Arrays.asList(singleMethod.getGroups()));
         }
 
-        this.sSSLink = "file:" + File.separator + File.separator + sOutputDir + File.separator + "html"
-                + File.separator + singleMethod.getId() + ".html";
+        this.sSSMsg = makeContentForLinksColumn(iTestResult);
 
         // if failed, then add error details
         if (iStatus == ITestResult.FAILURE) {
-            lstError.add(iTestResult.getThrowable().toString());
+            String stacktrace = StringUtils.defaultString(StringUtils.substringBetween(
```

Comment:
I would suggest that you please break this down into individual calls. It becomes difficult to read and also difficult to debug as well.